src/accommodation/accommodation.service.ts

```diff
@@ -327,12 +328,22 @@ export class AccommodationService {
       };
     }
 
-    if (checkInDate && checkOutDate) {
-      findManyOptions.where = {
-        ...findManyOptions.where,
-        AND: [{ availableFrom: { lte: checkInDate } }, { availableTo: { gte: checkOutDate } }],
-      };
+    if (this.isInvalidDateRange(checkInDate, checkOutDate)) {
+      throw new BadRequestException(ErrorsTypes.BAD_REQUEST_INVALID_DATE_RANGE);
     }
+
+    findManyOptions.where = {
+      ...findManyOptions.where,
+      AND: [{ availableFrom: { lte: checkInDate } }, { availableTo: { gte: checkOutDate } }],
+    };
+  }
+
+  private isInvalidDateRange(checkIn: Date | undefined, checkOut: Date | undefined) {
+    const result =
+      (!checkIn && checkOut) ||
+      (checkIn && !checkOut) ||
+      (checkIn && checkOut && dayjs(checkOut).isSameOrBefore(dayjs(checkIn)));
```

Comment:
doesn't enough here       !checkIn || !checkOut || dayjs(checkOut).isSameOrBefore(dayjs(checkIn)) ?

Reply:
in this case, it will give error when there is no check-in, regardless there is check-out or not.
But I need it to give error only when there is no checkin while there is checkout; or vice versa.

If both are missing, it is acceptable, i will just send all the accommodations